io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -563,9 +574,17 @@ public boolean hasNext() {
 		 *         number of results reported by the WFS.
 		 */
 		protected boolean isFeatureLimitReached() {
+			boolean condition = (maxNumberOfFeatures != UNLIMITED
+					&& totalFeaturesProcessed >= maxNumberOfFeatures)
+					|| (size != UNKNOWN_SIZE && totalFeaturesProcessed >= size)
+					|| (totalFeaturesProcessed == size && !iterator.hasNext());
```

Comment:
Why is there a change to the condition needed with this additional term?

Reply:
Because the additional objects found after the last main instance would be lost.

Reply:
Thank you. I think it would be very valuable to have that as a comment on that term.